src/core/hid/emulated_controller.cpp

```diff
@@ -866,7 +866,49 @@ void EmulatedController::SetLedPattern() {
     }
 }
 
+void EmulatedController::SetSupportedNpadStyleTag(NpadStyleTag supported_styles) {
+    supported_style_tag = supported_styles;
+    if (!is_connected) {
+        return;
+    }
+    if (!IsControllerSupported()) {
+        LOG_ERROR(Service_HID, "Controller type is not supported. Disconnecting controller");
+        Disconnect();
+    }
+}
+
+bool EmulatedController::IsControllerSupported() const {
+    switch (npad_type) {
+    case Core::HID::NpadStyleIndex::ProController:
```

Comment:
```suggestion
    case NpadStyleIndex::ProController:
```

Given this class is in the same namespace. Ditto for the other entries in the switch